app/assets/javascripts/views/shared/chartWidgetView.js

```diff
@@ -300,6 +301,7 @@
       if(axis == 'X'){ this.options.xLabel = value;}
       else { this.options.yLabel = value;}
       this._renderChart();
```

Comment:
Theres a double state update going on here `this._renderChart` triggers an update on the widget's `state.config.widgets[index]`. Then you manually update it again in the callback. Instead of a callback you could update the options first and call this._renderChart later. Besides this little issue seems good to go